pkg/operator/ceph/csi/peermap/config.go

```diff
@@ -0,0 +1,337 @@
+
+	err = clusterContext.Client.Get(ctx, request, existingConfigMap)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			// Create new configMap
+			return createConfig(clusterContext, request, data)
+		}
+		return errors.Wrapf(err, "failed to get existing mapping configMap %q", existingConfigMap.Name)
+	}
+
+	// Update existing configMap
+	existingConfigMap.Data[mappingConfigkey] = data
+	if err := clusterContext.Client.Update(ctx, existingConfigMap); err != nil {
+		return errors.Wrapf(err, "failed to update existing mapping configMap %q", existingConfigMap.Name)
+	}
+
+	return nil
+}
+
+func createConfig(clusterContext *clusterd.Context, request types.NamespacedName, data string) error {
+	newConfigMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      request.Name,
+			Namespace: request.Namespace,
+		},
+		Data: map[string]string{
+			mappingConfigkey: data,
+		},
+	}
+
+	// Get Operator owner reference
+	operatorPodName := os.Getenv(k8sutil.PodNameEnvVar)
+	ownerRef, err := k8sutil.GetDeploymentOwnerReference(clusterContext.Clientset, operatorPodName, request.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to get operator owner reference")
+	}
+	if ownerRef != nil {
+		blockOwnerDeletion := false
+		ownerRef.BlockOwnerDeletion = &blockOwnerDeletion
+	}
+
+	ownerInfo := k8sutil.NewOwnerInfoWithOwnerRef(ownerRef, request.Namespace)
+
+	// Set controller reference only when creating the configMap for the first time
+	err = ownerInfo.SetControllerReference(newConfigMap)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference on configMap %q", newConfigMap.Name)
+	}
+
+	err = clusterContext.Client.Create(context.TODO(), newConfigMap)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create mapping configMap %q", newConfigMap.Name)
+	}
+	return nil
+}
+
+func decodePeerToken(token string) (*cephclient.PeerToken, error) {
+	// decode the base64 encoded token
+	decodedToken, err := base64.StdEncoding.DecodeString(string(token))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode bootstrap peer token")
+	}
+
+	// Unmarshal the decoded token to a Go type
+	var decodedTokenToGo cephclient.PeerToken
+	err = json.Unmarshal(decodedToken, &decodedTokenToGo)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal decoded token")
+	}
+
+	logger.Debugf("peer cluster info %+v", decodedTokenToGo)
+
+	return &decodedTokenToGo, nil
+}
+
+func createTmpFile(content, pattern string) (*os.File, error) {
+	// Create tmp file path
+	filePath, err := ioutil.TempFile("/tmp", pattern)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create tmp file to store keyring")
+	}
+
+	// Write content into a file
+	err = ioutil.WriteFile(filePath.Name(), []byte(content), 0400)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to write token to file %q", filePath.Name())
+	}
+
+	return filePath, nil
+}
+
+func removeTmpFile(path string) {
```

Comment:
This function does not seem needed since it does a single thing only.

Reply:
removed the wrapping method